src/api/app/controllers/webui/users/notifications_controller.rb

```diff
@@ -1,10 +1,23 @@
 class Webui::Users::NotificationsController < Webui::WebuiController
-  before_action :require_login
   MAX_PER_PAGE = 300
+  VALID_NOTIFICATION_TYPES = ['read', 'reviews', 'comments', 'requests', 'unread'].freeze
+
+  before_action :require_login
+  before_action :check_param_type, :check_param_project, only: :index
 
   def index
     notifications_for_subscribed_user = NotificationsFinder.new.for_subscribed_user
-    @notifications = NotificationsFinder.new(notifications_for_subscribed_user).for_notifiable_type(params[:type])
+    @notifications = if params[:project]
+                       NotificationsFinder.new(notifications_for_subscribed_user).for_project_name(params[:project])
+                     else
+                       NotificationsFinder.new(notifications_for_subscribed_user).for_notifiable_type(params[:type])
+                     end
+
+    @projects = Project.joins(:notifications)
+                       .where(notifications: { subscriber: User.session, delivered: false })
+                       .order('name desc').group(:name).count
+                       .sort_by(&:last).reverse.to_h
```

Comment:
My only improvement is to move this four lines into a method with a descriptive name so we can know at a quick glance what this is doing.

Reply:
@danidoni thanks! It's been moved to the method `projects_for_filter`.